### PR TITLE
fix modal container

### PR DESCRIPTION
**Pull request by velrest**
The documentation states that the `container` argument can be set to a string
which will then be used as the container query string. Until now the argument
was ignored and just set to the `rootElement` or `body`.

### CODE REVIEW

**Review by anehx — APPROVED** [review]

File: addon/components/uk-modal.js [hunk 1]
```diff
@@ -1,58 +1,74 @@
...
-  bgClose: true,
-  stack: false,
-  container: true,
-  clsPage: "uk-modal-page",
-  clsPanel: "uk-modal-dialog",
-  selClose: [
+export default class UkModal extends Component {
+  layout = layout;
+
+  modalId = `modal-${guidFor(this)}`;
```
Comment (anehx) [inline addon/components/uk-modal.js:16]: Might be better to use a uuid so it's future proof (glimmer)

File: addon/components/uk-modal.js [hunk 2]
```diff
@@ -1,58 +1,74 @@
...
-      .lookup("service:-document")
-      .querySelector(this.container);
-  }),
+  get containerSelector() {
+    // Only set the container to the default if no query string was passed as argument.
+    if (typeof this.container !== "string") {
+      return getOwner(this).rootElement || "body";
+    }
+
+    return this.container;
```
Comment (anehx) [inline addon/components/uk-modal.js:54]:
```suggestion
    return this.args.container;
```
Right?

File: addon/components/uk-modal.js [hunk 3]
```diff
@@ -77,77 +93,96 @@ export default Component.extend({
...
-    );
+    super.didInsertElement(...args);
+    this.modal = UIkit.modal(`#${this.modalId}`, {
+      escClose: this.escClose,
+      bgClose: this.bgClose,
+      stack: this.stack,
+      container: this.containerSelector,
+      clsPage: this.clsPage,
+      clsPanel: this.clsPanel,
+      selClose: this.selClose,
```
Comment (anehx) [inline addon/components/uk-modal.js:123]: Shouldn't all of those come from `this.args` ?

File: addon/components/uk-modal.js [hunk 4]
```diff
@@ -77,77 +93,96 @@ export default Component.extend({
...
+
+      // Short-circuit if no mutations match the filter
+      if (!mutations.length) {
+        return;
+      }
+
+      const modalIsOpen = mutations.every((classList) =>
+        classList.contains("uk-open")
       );
+      this.focusTrapActive = Boolean(modalIsOpen);
```
Comment (anehx) [inline addon/components/uk-modal.js:170]: `.every` already returns a boolean

File: package.json [hunk 5]
```diff
@@ -104,9 +104,6 @@
     "qunit-dom": "1.6.0",
     "sass": "1.32.8"
   },
-  "resolutions": {
-    "focus-trap": "6.2.1"
-  },
```
Comment (anehx) [inline package.json:109]: Also remove the `focus-trap` dependency and update the lockfile

**Review by anehx — CHANGES_REQUESTED** [review]

File: addon/components/uk-modal.js (hunk 2 above)
Comment (velrest) [reply addon/components/uk-modal.js:54]: In `@ember/component` the `args` is undefined.

Comment (anehx) [reply addon/components/uk-modal.js:54]: Right, in that case I'll resolve those..

File: package.json (hunk 5 above)
Comment (velrest) [reply package.json:109]: Done :+1:

File: addon/components/uk-modal.js (hunk 1 above)
Comment (velrest) [reply addon/components/uk-modal.js:16]: is glimmer not playing well with `guidFor`?

Comment (anehx) [reply addon/components/uk-modal.js:16]: > You can call this on any object, EmberObject-based or not.

According to the docs this only works on ember objects which [a glimmer component isn't](https://github.com/glimmerjs/glimmer.js/blob/master/packages/%40glimmer/component/addon/-private/component.ts#L142).

Comment (velrest) [reply addon/components/uk-modal.js:16]: EmberObject-based of **not**.

According to the api docs this is what is accepted for first argument:
> obj Object any object, string, number, Element, or primitive

https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Finternals/guidFor

Comment (anehx) [reply addon/components/uk-modal.js:16]: :rofl: reading helps. NVM then. Now you just have to resolve conflicts and we're good to go

**Comment by anehx** [comment]
Closes #653

**Review by anehx — APPROVED** [review]

**Pull request merged**